### PR TITLE
fix: patch for std4#195 (more succ/pred lemmas for `Nat`)

### PR DESCRIPTION
**Pull request by fgdorais**
---

- [x] depends on: https://github.com/leanprover/std4/pull/195

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/Nat/Basic.lean [hunk 1]
```diff
@@ -165,7 +165,6 @@ theorem eq_of_le_of_lt_succ {n m : ℕ} (h₁ : n ≤ m) (h₂ : m < n + 1) : m
 -- Moved to Std
 #align nat.one_add Nat.one_add
 
-@[simp]
```
Comment (semorrison) [inline Mathlib/Data/Nat/Basic.lean:168]: Is this `@[simp]` annotation useless / disallowed / redundant?

File: Mathlib/Init/Data/Nat/Lemmas.lean [hunk 2]
```diff
@@ -784,7 +784,7 @@ lemma to_digits_core_length (b : Nat) (h : 2 <= b) (f n e : Nat)
             to_digits_core_lens_eq b f (n / b) (Nat.digitChar $ n % b), if_false]
           exact Nat.succ_le_succ ih
       case neg =>
-        have _ : e = 0 := Nat.eq_zero_of_nonpos e h_pred_pos
+        have _ : e = 0 := Nat.eq_zero_of_not_pos h_pred_pos
```
Comment (semorrison) [inline Mathlib/Init/Data/Nat/Lemmas.lean:787]: Isn't this contrary to the naming conventions?

File: Mathlib/Combinatorics/Hindman.lean [hunk 3]
```diff
@@ -244,7 +244,7 @@ set_option linter.uppercaseLean3 false in
 theorem FP_drop_subset_FP {M} [Semigroup M] (a : Stream' M) (n : ℕ) : FP (a.drop n) ⊆ FP a := by
   induction' n with n ih
   · rfl
-  rw [Nat.succ_eq_one_add, ← Stream'.drop_drop]
+  rw [← Nat.one_add, ← Stream'.drop_drop]
```
Comment (semorrison) [inline Mathlib/Combinatorics/Hindman.lean:247]: @fgdorais is about to undo all of these.

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Data/Nat/Basic.lean (hunk 1 above)
Comment (fgdorais) [reply Mathlib/Data/Nat/Basic.lean:168]: I think this was demanded by the linter, so it's one of the three options.

**Pull request merged**